niwidgets/streamlines.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import print_function
+import nibabel as nib
+import numpy as np
+from ipywidgets import interact, fixed, widgets
+import os
+import ipyvolume as ipv
+
+
+def length(x):
+    """Returns the sum of euclidean distances between neighboring points"""
+    return np.sum(np.sqrt(np.sum((x[:-1, :] - x[1:, :]) *
+                                 (x[:-1, :] - x[1:, :]),
+                                 axis=1)))
+
+
+def color(x):
+    """Returns an approximation for color of line based on its endpoints"""
+    dirvec = (x[0, :] - x[-1, :])
+    return (dirvec/(np.sqrt(np.sum(dirvec*dirvec, axis=-1)))).dot(np.eye(3))
+
+
+class StreamlineWidget:
+    """
+    Turns mrtrix track or trackvis files into interactive plots using ipyvolume.
+
+    Color for each line is rendered as a function of the endpoints of the
+    streamline. Currently, this resolves to red for left-right, green for
+    anterior-posterior, and blue for inferior-superior.
+
+    Args
+    ----
+        filename : str
+                The path to your ``.trk`` file. Can be a string, or a
+                ``PosixPath`` from python3's pathlib.
```

Comment:
PosixPath doesn't work for me. Raises: 

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-09a88a08f323> in <module>()
----> 1 sw = StreamlineWidget(filename=streamlines)
      2 
      3 style = {'axes': {'color': 'red',
      4                   'label': {'color': 'white'},
      5                   'ticklabel': {'color': 'white'},

~/source/niwidgets/niwidgets/streamlines.py in __init__(self, filename, streamlines)
     41 
     42         if filename:
---> 43             if not os.path.isfile(filename):
     44                 # for Python3 should have FileNotFoundError here
     45                 raise IOError('file {} not found'.format(filename))

~/anaconda3/lib/python3.5/genericpath.py in isfile(path)
     28     """Test whether a path is a regular file"""
     29     try:
---> 30         st = os.stat(path)
     31     except OSError:
     32         return False

TypeError: argument should be string, bytes or integer, not PosixPath
```

Reply:
i'm curious as to your environment:

on my osx with py36 i get this:

```python
In [1]: from niwidgets.exampledata import streamlines

In [2]: streamlines
Out[2]: PosixPath('/software/nipy-repo/niwidgets/niwidgets/data/streamlines.trk')

In [3]: from niwidgets.streamlines import StreamlineWidget

In [4]: sw = StreamlineWidget(filename=streamlines)
```